src/Symfony/Component/Workflow/DefinitionBuilder.php

```diff
@@ -79,15 +79,36 @@ public function addPlaces(array $places)
         }
     }
 
+    /**
+     * @param (Transition|array)[] $transitions
+     */
     public function addTransitions(array $transitions)
     {
         foreach ($transitions as $transition) {
-            $this->addTransition($transition);
+            if ($transition instanceof Transition) {
+                $this->addTransition($transition);
+            } else {
+                list($name, $froms, $tos) = $transition;
+                $this->createTransition($name, $froms, $tos);
+            }
         }
     }
 
+    /**
+     * @param Transition $transition
+     */
     public function addTransition(Transition $transition)
     {
         $this->transitions[] = $transition;
     }
+
+    /**
+     * @param string          $name
+     * @param string[]|string $froms
+     * @param string[]|string $tos
+     */
+    public function createTransition($name, $froms, $tos)
```

Comment:
the method name is misleading, because it does not only create the Transition (which would return it for usage) but instead it creates it internally and then adds it